src/auto/dbus_connection.rs

```diff
@@ -0,0 +1,1264 @@
+    ) -> Pin<
+        Box_<
+            dyn std::future::Future<Output = Result<(glib::Variant, UnixFDList), glib::Error>>
+                + 'static,
+        >,
+    > {
+        let bus_name = bus_name.map(ToOwned::to_owned);
+        let object_path = String::from(object_path);
+        let interface_name = String::from(interface_name);
+        let method_name = String::from(method_name);
+        let parameters = parameters.map(ToOwned::to_owned);
+        let reply_type = reply_type.map(ToOwned::to_owned);
+        let fd_list = fd_list.map(ToOwned::to_owned);
+        Box_::pin(crate::GioFuture::new(self, move |obj, send| {
+            let cancellable = Cancellable::new();
+            obj.call_with_unix_fd_list(
+                bus_name.as_ref().map(::std::borrow::Borrow::borrow),
+                &object_path,
+                &interface_name,
+                &method_name,
+                parameters.as_ref().map(::std::borrow::Borrow::borrow),
+                reply_type.as_ref().map(::std::borrow::Borrow::borrow),
+                flags,
+                timeout_msec,
+                fd_list.as_ref().map(::std::borrow::Borrow::borrow),
+                Some(&cancellable),
+                move |res| {
+                    send.resolve(res);
+                },
+            );
+
+            cancellable
+        }))
+    }
+
+    #[cfg(any(unix, feature = "dox"))]
+    pub fn call_with_unix_fd_list_sync<P: IsA<UnixFDList>, Q: IsA<Cancellable>>(
+        &self,
+        bus_name: Option<&str>,
+        object_path: &str,
+        interface_name: &str,
+        method_name: &str,
+        parameters: Option<&glib::Variant>,
+        reply_type: Option<&glib::VariantTy>,
+        flags: DBusCallFlags,
+        timeout_msec: i32,
+        fd_list: Option<&P>,
+        cancellable: Option<&Q>,
+    ) -> Result<(glib::Variant, UnixFDList), glib::Error> {
+        unsafe {
+            let mut out_fd_list = ptr::null_mut();
+            let mut error = ptr::null_mut();
+            let ret = gio_sys::g_dbus_connection_call_with_unix_fd_list_sync(
+                self.to_glib_none().0,
+                bus_name.to_glib_none().0,
+                object_path.to_glib_none().0,
+                interface_name.to_glib_none().0,
+                method_name.to_glib_none().0,
+                parameters.to_glib_none().0,
+                reply_type.to_glib_none().0,
+                flags.to_glib(),
+                timeout_msec,
+                fd_list.map(|p| p.as_ref()).to_glib_none().0,
+                &mut out_fd_list,
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                &mut error,
+            );
+            if error.is_null() {
+                Ok((from_glib_full(ret), from_glib_full(out_fd_list)))
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    pub fn close<P: IsA<Cancellable>, Q: FnOnce(Result<(), glib::Error>) + Send + 'static>(
```

Comment:
Is there a mechanism to "open" a connection? If not, I think it should take `self`.

Reply:
as in "reopen"? no.

Is there a way to make it take `self`? Without manual code I mean >_<

Reply:
I'm not sure. You can update an argument type but not sure about `self`... Any idea @EPashkin ?

Reply:
`self` makes no sense here because it's all reference counted. Just like the other `close()` functions we have.

This is ok as is.